src/sidebar/store/modules/annotations.ts

```diff
@@ -80,73 +111,44 @@ function findByTag(annotations, tag) {
  * `annotation` may either be new (unsaved) or a persisted annotation retrieved
  * from the service.
  *
- * @param {Omit<Annotation, '$anchorTimeout'>} annotation
- * @param {string} tag - The `$tag` value that should be used for this if it
- *                       doesn't have a `$tag` already
- * @param {string|null} currentUserId - The account id of the currently-auth'd
- *                       user, if any
- * @return {Annotation} - API annotation data with client annotation data merged
+ * @param tag - The `$tag` value that should be used for this if it doesn't have
+ * a `$tag` already
+ * @return - API annotation data with client annotation data merged
  */
-function initializeAnnotation(annotation, tag, currentUserId) {
+function initializeAnnotation(
+  annotation: Omit<Annotation, '$anchorTimeout'>,
+  tag: string,
+  currentUserId: string | null
+): Annotation {
   let orphan = annotation.$orphan;
 
   if (!annotation.id) {
-    // New annotations must be anchored
+    // Unsaved (new) annotations must be anchored
     orphan = false;
   }
 
-  let $cluster = /** @type {HighlightCluster} */ ('other-content');
+  let $cluster: HighlightCluster = 'other-content';
   if (annotation.user === currentUserId) {
     $cluster = isHighlight(annotation) ? 'user-highlights' : 'user-annotations';
   }
 
-  return Object.assign(
-    {},
-    annotation,
-    /** @type {ClientAnnotationData} */ ({
-      $anchorTimeout: false,
-      $cluster,
-      $tag: annotation.$tag || tag,
-      $orphan: orphan,
-    })
-  );
+  return Object.assign({}, annotation, {
+    $anchorTimeout: false,
+    $cluster,
+    $tag: annotation.$tag || tag,
+    $orphan: orphan,
+  });
 }
 
-const initialState = {
-  /**
-   * Set of all currently loaded annotations.
-   *
-   * @type {Annotation[]}
-   */
-  annotations: [],
-  /**
-   * Annotations whose cards or highlights are currently hovered.
-   *
-   * The styling of the highlights/cards of these annotations are adjusted to
-   * show the correspondence between the two.
-   *
-   * @type {Record<string, boolean>}
-   */
-  hovered: {},
-  /**
-   * A map of annotations that should appear as "highlighted", e.g. the
-   * target of a single-annotation view
-   *
-   * @type {Record<string, boolean>}
-   */
-  highlighted: {},
-  /** The local tag to assign to the next annotation that is loaded into the app. */
-  nextTag: 1,
-};
-
-/** @typedef {typeof initialState} State */
-
 const reducers = {
-  /**
-   * @param {State} state
-   * @param {{ annotations: Annotation[], currentAnnotationCount: number, currentUserId: string|null }} action
-   */
-  ADD_ANNOTATIONS(state, action) {
+  ADD_ANNOTATIONS(
+    state: State,
+    action: {
+      annotations: Annotation[];
+      currentAnnotationCount: number;
+      currentUserId: string | null;
+    }
+  ): Partial<State> {
```

Comment:
It's true that all of these actions return `Partial<State>` but there might be a more efficient way of annotating that.

Reply:
It isn't necessary provide this annotation at all, but `Partial<State>` seems fine if added for readability. An alternative would be to put one assertion at the end using `satisfies` - eg. `const reducers =  { .... } satisfies Record<string, Partial<State>>` (see [docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator)). This is similar to `as` but doesn't "erase" the specific information about the keys and values.

Reply:
ha-HAH! This achieves my goal of "find some way to use `satisfies` by the end of the week." `satisfies` was only added in 4.9 in November, but this project is at 4.9.4 so we're fine...

Reply:
@robertknight Note that `reducers` maps an action (name) to a function. This syntax seems to work.

`} satisfies Record<string, (...args: any[]) => Partial<State>>;`

Could narrow arg types more if we cared to (the first arg if present is always `State`), but I'm not sure that really adds anything but noise.